wakuv2/waku.go

```diff
@@ -260,48 +264,86 @@ func New(nodeKey string, cfg *Config, logger *zap.Logger, appdb *sql.DB) (*Waku,
 	return waku, nil
 }
 
-func (w *Waku) addPeers(addresses []string, protocol libp2pproto.ID) {
+func (w *Waku) addPeers(addresses []string, protocol libp2pproto.ID, fnForEachPeer func(ma multiaddr.Multiaddr, protocol libp2pproto.ID)) {
 	for _, addrString := range addresses {
 		if addrString == "" {
 			continue
 		}
 
-		addr, err := multiaddr.NewMultiaddr(addrString)
-		if err != nil {
-			log.Warn("invalid peer multiaddress", addrString, err)
-			continue
+		if strings.HasPrefix(addrString, "enrtree://") {
+			// Use DNS Discovery
+			go w.dnsDiscover(addrString, protocol, fnForEachPeer)
+		} else {
+			// It's a normal multiaddress
+			w.addPeerFromString(addrString, protocol, fnForEachPeer)
 		}
+	}
+}
+
+func (w *Waku) dnsDiscover(enrtreeAddress string, protocol libp2pproto.ID, apply func(ma multiaddr.Multiaddr, protocol libp2pproto.ID)) {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	w.dnsAddressCacheLock.RLock()
+	multiaddresses, ok := w.dnsAddressCache[enrtreeAddress]
```

Comment:
I "think" your cache will always be empty, I cannot see where you populate it

Reply:
oops!